TIcTacToe/tic-tac-toe/src/contexts/GameContext.js

```diff
@@ -19,7 +19,7 @@ export function GameProvider({ children }) {
   );
 }
 
-export const usePlayUserTurn = () => {
+export const usePlayUserTurn = (i) => {
```

Comment:
this `i` is not used right now. (you can look for variable shadowing) Chose one of below implementations (I would prefer first one):

```js
const usePlayUserTurn = (index) => {
  // ...
  const playUserTurn = () => {
    // ...
    if (winner || squares[index]) return;
    squares[index] = xIsNext ? "X" : "O";
    // ...
  };
  // ...
}
// ...
const playUserTurn = usePlayUserTurn(index);
// ...
onClick={playUserTurn}
```
or

```js
const usePlayUserTurn = () => {
  // ...
  const playUserTurn = (index) => {
    // ...
    if (winner || squares[index]) return;
    squares[index] = xIsNext ? "X" : "O";
    // ...
  };
  // ...
}
// ...
const playUserTurn = usePlayUserTurn();
// ...
onClick={() => playUserTurn(index)}

```